clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperation.java

```diff
@@ -472,28 +473,75 @@ private void scaleApplication(
         convertToMb("memory", description.getApplicationAttributes().getMemory());
     Integer diskSizeAmount =
         convertToMb("disk quota", description.getApplicationAttributes().getDiskQuota());
-
     client
         .getProcesses()
         .scaleProcess(
             serverGroupId,
             description.getApplicationAttributes().getInstances(),
             memoryAmount,
             diskSizeAmount);
+
+    if (!description.getApplicationAttributes().getProcesses().isEmpty()) {
+      List<Process> processes = client.getProcesses().getAllProcessesByAppId(serverGroupId);
+
+      for (ProcessRequest req : description.getApplicationAttributes().getProcesses()) {
+        String processGuid =
+            processes.stream()
+                .filter(p -> p.getType().equalsIgnoreCase(req.getType()))
+                .map(p -> p.getGuid())
+                .findFirst()
+                .orElseThrow(
+                    () ->
+                        new CloudFoundryApiException(
+                            "Unable to find a process with type: " + req.getType()));
```

Comment:
Should this fail for _all_ processes when one of them is not found?

Reply:
I think so, yes. In the scenario you mention, if someone is deploying an app to production and it requires two processes to have specific commands, if one of them gets skipped because it wasn't found it could be problematic and they might not even know that it was skipped. In this case, we want to enforce correct manifests and specifically correct process requests.